numba/tests/test_parfors.py

```diff
@@ -2686,6 +2707,26 @@ def test_impl():
 
         self.prange_tester(test_impl)
 
+    @skip_parfors_unsupported
+    def test_multi_defined_array_index(self):
+        # issue6597
+        def test_impl(a, b, c):
+            n_rows = b.shape[0]
+            for x in range(1):
+                i_row = 0
+                for _ in range(n_rows):
+                    bi = b[i_row]
+                    c[:] = a[i_row, :bi]
+                    i_row += 1
+                    if i_row >= n_rows:
+                        i_row -= n_rows
+            return c
+
+        a = np.arange(10).reshape((10, 1))
+        b = np.ones(10, dtype=np.int)
+        c = np.ones(1, dtype=np.int)
```

Comment:
Suspect this will fail on windows as `np.int` is aliased to pythons `int`, think this will end up as a 32bit int array which is probably the root cause of the typing failures on windows CI. Suggest using `dtype=np.int64`.